ledger-service/pureconfig-utils/src/main/scala/com/daml/pureconfigutils/SharedConfigReaders.scala

```diff
@@ -6,63 +6,129 @@ package com.daml.pureconfigutils
 import akka.http.scaladsl.model.Uri
 import com.auth0.jwt.algorithms.Algorithm
 import com.daml.dbutils.JdbcConfig
-import com.daml.jwt.{ECDSAVerifier, HMAC256Verifier, JwksVerifier, JwtVerifierBase, RSA256Verifier}
+import com.daml.jwt.{
+  ECDSAVerifier,
+  HMAC256Verifier,
+  JwksVerifier,
+  JwtVerifier,
+  JwtVerifierBase,
+  RSA256Verifier,
+}
+import com.daml.ledger.api.tls.TlsConfiguration
+import com.daml.metrics.MetricsReporter
 import com.daml.platform.services.time.TimeProviderType
-import pureconfig.{ConfigReader, ConvertHelpers}
+import pureconfig.error.{CannotConvert, ConvertFailure, FailureReason}
+import pureconfig.{ConfigObjectCursor, ConfigReader, ConvertHelpers}
 import pureconfig.generic.semiauto.deriveReader
+import scalaz.{-\/, \/, \/-}
 
 import java.nio.file.Path
+import java.io.File
 import scala.concurrent.duration.FiniteDuration
 
 final case class HttpServerConfig(address: String, port: Int, portFile: Option[Path] = None)
-final case class LedgerApiConfig(address: String, port: Int)
-final case class MetricsConfig(reporter: String, reportingInterval: FiniteDuration)
+final case class LedgerTlsConfig(
+    enabled: Boolean = false,
+    certChainFile: Option[File] = None,
+    privateKeyFile: Option[File] = None,
+    trustCollectionFile: Option[File] = None,
+) {
+  def tlsConfiguration: TlsConfiguration =
+    TlsConfiguration(enabled, certChainFile, privateKeyFile, trustCollectionFile)
+}
+final case class LedgerApiConfig(
+    address: String,
+    port: Int,
+    tls: LedgerTlsConfig = LedgerTlsConfig(),
+)
+final case class MetricsConfig(reporter: MetricsReporter, reportingInterval: FiniteDuration)
+
+object TokenVerifierConfig {
+  val knownTokenVerifiers =
+    List("hs256-unsafe", "rs256-crt", "es256-crt", "es512-crt", "rs256-jwks")
```

Comment:
I really like this idea, but I want to take it further to guarantee consistency between the error message and the actual set of verifiers, and also to not treat the first element of this list magically. That is done in 71680e16a251cffd32556822e97610c93078c905 in branch `hocon_json_api+s11`.